lgc/state/PalMetadata.cpp

```diff
@@ -205,6 +205,22 @@ void PalMetadata::mergeFromBlob(StringRef blob) {
             *destNode = std::min(destNode->getUInt(), srcNode.getUInt());
             return 0;
           }
+          if (mapKey.getString() == Util::Abi::HardwareStageMetadataKey::SgprCount) {
+            *destNode = std::max(destNode->getUInt(), srcNode.getUInt());
+            return 0;
+          }
+          if (mapKey.getString() == Util::Abi::HardwareStageMetadataKey::SgprLimit) {
+            *destNode = std::max(destNode->getUInt(), srcNode.getUInt());
+            return 0;
+          }
+          if (mapKey.getString() == Util::Abi::HardwareStageMetadataKey::VgprCount) {
+            *destNode = std::max(destNode->getUInt(), srcNode.getUInt());
+            return 0;
+          }
+          if (mapKey.getString() == Util::Abi::HardwareStageMetadataKey::VgprLimit) {
+            *destNode = std::max(destNode->getUInt(), srcNode.getUInt());
+            return 0;
```

Comment:
Weird... I had a comment here that wasn't submitted with the review.

The point was that since all of these fields do the same thing for the same reason, you could just merge those four if-statements.

Reply:
done.